runtime/src/subject_alt_name.rs

```diff
@@ -0,0 +1,60 @@
+/*++
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    populate_idev.rs
+
+Abstract:
+
+    File contains AddSubjectAltName mailbox command.
+
+--*/
+
+use core::str::from_utf8;
+
+use arrayvec::ArrayVec;
+use caliptra_common::mailbox_api::{AddSubjectAltNameReq, MailboxResp};
+use caliptra_error::{CaliptraError, CaliptraResult};
+use zerocopy::AsBytes;
+
+use crate::{Drivers, MAX_CERT_CHAIN_SIZE, PL0_PAUSER_FLAG};
+
+pub struct AddSubjectAltNameCmd;
+impl AddSubjectAltNameCmd {
+    // https://www.dmtf.org/sites/default/files/standards/documents/DSP0274_1.3.0.pdf 1.3.6.1.4.1.412.274.1
+    pub const DMTF_OID: &'static [u8] =
+        &[0x2B, 0x06, 0x01, 0x04, 0x01, 0x83, 0x1C, 0x82, 0x12, 0x01];
+
+    pub(crate) fn execute(drivers: &mut Drivers, cmd_args: &[u8]) -> CaliptraResult<MailboxResp> {
+        if cmd_args.len() <= core::mem::size_of::<AddSubjectAltNameReq>() {
+            let mut cmd = AddSubjectAltNameReq::default();
+            cmd.as_bytes_mut()[..cmd_args.len()].copy_from_slice(cmd_args);
+
+            let dmtf_device_info_size = cmd.dmtf_device_info_size as usize;
+            if dmtf_device_info_size > cmd.dmtf_device_info.len() {
+                return Err(CaliptraError::RUNTIME_MAILBOX_INVALID_PARAMS);
+            }
+
+            // validate DMTF otherName
+            let dmtf_device_info_utf8 =
```

Comment:
nit: Might be worth moving to a private helper just to make it clear that we're just validating the input rather than doing any conversion

Reply:
Done